ScintSuite/SimulationCodes/SINPA/_Forw_Mod_xarray.py

```diff
@@ -0,0 +1,1713 @@
+                r = ssPH.sel(x=pitch, y=gyro, method = 'nearest').y.item()
+                # Fill the matrices for pinhole (PH) and scintillator (SC)
+                ssPH.loc[p,r] += weight/p_step/r_step
+                ssSC += (WF.sel(x=p, y=r, drop=True))\
+                                *weight*(2*np.pi/gyrophases)
+                # Control of the signal lost
+                ions_pinhole_lost -= weight
+                ions_pinhole_found += weight                
+
+    synthetic_signal['PH'] = ssPH
+    synthetic_signal['SC'] = ssSC
+
+
+    print("ions/s considered for the synth. sig. = "+str(f"{ions_pinhole:e}"))
+    print("ions/s analysed = "+str(f"{ions_pinhole_found:e}"))    
+    print("ions/s lost = "+str(f"{ions_pinhole_lost:e}"))
+
+    # Default plot to control results
+    if plot == True:
+        fig, ax = plt.subplots(2,2, figsize=(8, 6),
+                                    facecolor='w', edgecolor='k') 
+        cmap = ss.plt.Gamma_II()
+        # Plot of the synthetic signals, pinhole and scintillator
+        ax_param = {'fontsize': 10, \
+                    'xlabel': 'Pitch [º]', 'ylabel': '$r_l [cm]$'}         
+        ssPH.transpose().plot.imshow(ax=ax[0,0],cmap=cmap,
+                                     vmax=0.5*ssPH.max().item(),
+                                     cbar_kwargs={"label": 'ions/(s cm deg)'})
+        ax[0,0] = ss.plt.axis_beauty(ax[0,0], ax_param)
+        ax[0,0].set_title("Pinhole")    
+        ssSC.transpose().plot.imshow(ax=ax[0,1], cmap=cmap,
+                                     vmax=0.5*ssPH.max().item(),
+                                     cbar_kwargs={"label": 'ions/(s cm deg)'})
+        ax[0,1] = ss.plt.axis_beauty(ax[0,1], ax_param)
+        ax[0,1].set_title("Scintillator")
+
+        # Plot of the distributions of pitch and gyroradius
+        ax_options_profiles = {'fontsize': 10, 'ylabel': 'Signal [a.u.]'}
+        (ssPH.sum(dim='y')/ssPH.sum(dim='y').integrate('x')).plot.\
+            line(ax=ax[1,0], color='black', label='Pinhole')
+        (ssSC.sum(dim='ys')/ssSC.sum(dim='ys').integrate('xs')).plot.\
+            line(ax=ax[1,0], color='red', label='Scintillator')
+        ax_options_profiles['xlabel'] = 'Pitch [$\\degree$]'  
+        ax[1,0] = ss.plt.axis_beauty(ax[1,0], ax_options_profiles)
+        ax[1,0].legend()        
+        (ssPH.sum(dim='x')/ssPH.sum(dim='x').integrate('y')).plot.\
+            line(ax=ax[1,1], color='black', label='Pinhole')
+        (ssSC.sum(dim='xs')/ssSC.sum(dim='xs').integrate('ys')).plot.\
+            line(ax=ax[1,1], color='red', label='Scintillator')      
+        ax_options_profiles['xlabel'] = 'Gyroradius [cm]'
+        ax[1,1] = ss.plt.axis_beauty(ax[1,1], ax_options_profiles)
+        ax[1,1].legend()
+
+        fig.tight_layout()
+        plt.show(block = False)
+
+    return synthetic_signal
+
+
+def pr_space_to_pe_space(synthetic_signal, B=4, A=2, Z=2, 
```

Comment:
I know. This is just a quick review of the results in case you want to check it. With xarray doing plots is easy enough so that each user can make it's own custom plot with the parameters they want to see. I think adding a "general" plotting function to make the plots easier for the user would be actually more difficult to use.